src/dagmc/dagmcmetadata.hpp

```diff
@@ -8,118 +8,316 @@
-  // delimeter<data>delimiter<data>delimiter into a vector of the form
-  // data[0],data[1] etc
+  /**
+   * @brief Unpacks a string into a vector of substrings.
+   *
+   * The input string is expected to be of the form
+   * "delimiter<data>delimiter<data>delimiter". The function will unpack this
+   * into a vector of the form "data[0],data[1],...".
+   *
+   * @param to_unpack The string to unpack.
+   * @param delimiters The delimiters used to separate the data in the string.
+   *                   Default is "|".
+   *
+   * @return Returns a vector of substrings extracted from the input string.
+   */
   std::vector<std::string> unpack_string(std::string to_unpack,
                                          std::string delimiters = "|");
 
-  // splits a string on the basis of the first delimiter it finds
+  /**
+   * @brief Splits a string on the basis of the first delimiter it finds.
+   *
+   * @param property_string The string to be split.
+   * @param delimiter The delimiter on which to split the string.
+   *
+   * @return Returns a pair of strings. The first element of the pair is the
+   *         substring before the first occurrence of the delimiter. The second
+   *         element is the substring after the first occurrence of the
+   * delimiter.
+   */
   std::pair<std::string, std::string> split_string(std::string property_string,
                                                    std::string delimiter);
 
-  // from a string of the form key:property/key:property
-  // return the value of a desired key
+  /**
+   * @brief Extracts the value of a desired key from a property string.
+   *
+   * The property string is expected to be of the form
+   * "key:property/key:property".
+   *
+   * @param property_string The string containing the properties.
+   * @param property The key for which to return the value.
+   * @param delimiter The delimiter used to separate the keys and values in the
+   *                  property string. Default is ":".
+   * @param chopped Optional. If true, the function will return the property
+   *                value after removing the key and delimiter. If false, it
+   *                will return the property value as is. Default is true.
+   *
+   * @return Returns the value of the specified key from the property string.
+   */
   std::string return_property(std::string property_string, std::string property,
                               std::string delimiter = ":", bool chopped = true);
 
-  // test to see if string is an int
+  /**
+   * @brief Tests if a string can be converted to an integer.
+   *
+   * @param value The string to be tested.
+   *
+   * @return Returns true if the string can be converted to an integer,
+   *         false otherwise.
+   */
   bool try_to_make_int(std::string value);
 
   // private member functions
  private:
-  // parse the material data
+  /**
+   * @brief Parses the material data from the associated DagMC instance.
```

Comment:
What does this actually do? Changes the state of some data structures in some way, but how? (And similarly for following methods)